retro_data_structures/formats/mrea.py

```diff
@@ -60,6 +61,81 @@ class MREAVersion(IntEnum):
     DonkeyKongCountryReturns = 0x20
 
 
+class AreaDependencyAdapter(Adapter):
+    def __init__(self, asset_id):
+        super().__init__(Struct(
+            Const(0, Int32ub),
+            "dependencies" / PrefixedArray(Int32ub, Struct(
+                asset_id=asset_id,
+                asset_type=FourCC,
+            )),
+            "offsets" / PrefixedArray(Int32ub, Int32ub),
+        ))
+
+    def _decode(self, obj, context, path) -> AreaDependencies:
+        layers = []
+        for i in range(len(obj.offsets) - 1):
+            start, finish = obj.offsets[i], obj.offsets[i+1]
+            layers.append([
+                Dependency(dep.asset_type, dep.asset_id)
+                for dep in obj.dependencies[start:finish]
+            ])
+
+        non_layer = [
+            Dependency(dep.asset_type, dep.asset_id)
+            for dep in obj.dependencies[obj.offsets[-1]:]
+        ]
+
+        return AreaDependencies(layers, non_layer)
+
+    def _encode(self, obj: AreaDependencies, context, path):
+        layer_deps = list(obj.layers)
+        layer_deps.append(obj.non_layer)
+
+        deps = []
+        offsets = []
+        for layer_dep in layer_deps:
+            offsets.append(len(deps))
+            deps.extend(
+                Container(asset_type=dep.type, asset_id=dep.id)
+                for dep in layer_dep
+                if not dep.exclude_for_mlvl
+            )
+
+        return {
+            "dependencies": deps,
+            "offsets": offsets
+        }
+
+
+class AreaModuleDependencyAdapter(Adapter):
+    def __init__(self):
+        super().__init__(Struct(
+            rel_module=PrefixedArray(Int32ub, String),
+            rel_offset=PrefixedArray(Int32ub, Int32ub),
+        ))
+
+    def _decode(self, obj, context, path) -> list[list[str]]:
+        offsets = list(zip(*[iter(obj.rel_offset)]*2))
```

Comment:
```python
    offset_iter = iter(obj.rel_offset)
    return [
        obj.rel_module[start:finish]
        for start, finish in zip(offset_iter, offset_iter)
    ]
```

and you complain about my poor ternary :P

Reply:
i *literally* took this from the python docs: https://docs.python.org/3/library/functions.html#zip

(i will change it)